packages/components/src/2016-12-01/button-state.js

```diff
@@ -0,0 +1,42 @@
+/**
+Copyright 2016 Autodesk,Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+const ButtonState = {
+  create(state = {}) {
+    return Object.assign({
+      disabled: false,
+      focus: false,
+      hover: false,
+      active: false,
+    }, state);
+  },
+
+  enterHover(state) {
+    return Object.assign({}, state, { hover: true });
```

Comment:
I haven't seen object-spread mixed with key-value pairs in a literal like that. Cool.